src/StructuredLogger/ObjectModel/IProjectOrEvaluation.cs

```diff
@@ -1,7 +1,77 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Configuration;
+
 namespace Microsoft.Build.Logging.StructuredLogger
 {
     public interface IProjectOrEvaluation
     {
-        string TargetFramework { get; set; }
+        public string AdornmentString { get; }
+
+        public string TargetFramework { get; set; }
+
+        public string Platform { get; set; }
+
+        public string Configuration { get; set; }
     }
-}
\ No newline at end of file
+
+    public static class IProjectOrEvaluationHelper
+    {
+        private class IProjectOrEvaluationComparer : IEqualityComparer<IProjectOrEvaluation>
+        {
+            public bool Equals(IProjectOrEvaluation x, IProjectOrEvaluation y)
+            {
+                return x.TargetFramework == y.TargetFramework
+                    && x.Platform == y.Platform
+                    && x.Configuration == y.Configuration;
+            }
+
+            public int GetHashCode(IProjectOrEvaluation obj)

```

Comment:
You can just take advantage of the ValueTuple hash:
public int GetHashCode(IProjectOrEvaluation obj) => (TargetFramework, Platform, Configuration).GetHashCode();

Reply:
Thanks, that worked.  My concern was dealing with nulls and empty string.